app/bot/openai/summary.go

```diff
@@ -0,0 +1,279 @@
+package openai
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-pkgz/lcw"
+	"github.com/go-pkgz/syncs"
+	tbapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
+)
+
+//go:generate moq --out mocks/openai_summary.go --pkg mocks --skip-ensure . openAISummary:OpenAISummary
+//go:generate moq --out mocks/remark.go --pkg mocks --skip-ensure . remarkGetComments:RemarkClient
+//go:generate moq --out mocks/ukeeper.go --pkg mocks --skip-ensure . uKeeperGet:UKeeperClient
+
+// Summarizer is a helper for summarizing articles by links
+type Summarizer struct {
+	openAISummary openAISummary
+	remark        remarkGetComments
+	uKeeper       uKeeperGet
+
+	cache lcw.LoadingCache
```

Comment:
adding comment to cache field explaining what the key and what the value usually helps to understand